lib/svtplay_dl/fetcher/hls.py

```diff
@@ -93,6 +94,7 @@ def hlsparse(options, res, url, **kwargs):
             streams[int(bit_rate)] = HLS(copy.copy(options), urls, bit_rate, cookies=res.cookies, keycookie=keycookie, authorization=authorization, audio=audio_url)
 
     elif m3u8.media_segment:
+        options.segments = False
```

Comment:
why do you set this?

Reply:
It is an unrelated bug fix that makes it possible to download hls with only the url to the m3u8 media_segment when it only contains one segment. Ex:
```
./svtplay-dl -S -M -f "https://mtgxse04-vh.akamaihd.net/i/open/20xxxxxx/2049xxxx/204952xx/20495279/5a30e587ea29838b8cc56e8f7c82d233_20495279_269270_20180222T213434_,150,300,600,1200,2400,.mp4.csmil/master.m3u8?__b__=2000&hdnts=st=1519667985~exp=3667151632~acl=/*~hmac=2e33027cf399db818ca38ef80d645ef3dbde1ed1ac3f773f04000b3c26a48386&cc1=name=Svenska%20f%C3%B6r%20h%C3%B6rselskadade~default=no~forced=no~lang=sv~uri=https%3A%2F%2Fsubstitch.play.mtgx.tv%2Fsubtitle%2Fconvert%2Fvtt%3Fsource%3Dhttps%3A%2F%2Fcdn-subtitles-mtgx-tv.akamaized.net%2Fpitcher%2F20xxxxxx%2F2049xxxx%2F204952xx%2F20495279%2F2800120619%2Funified.subs.20495279_sv_sdh.vtt%26output%3Dm3u8"
```

Reply:
Makes it possible to download m3u8 media_segments, Ex:
```
./svtplay-dl -S -M -f "https://mtgxse04-vh.akamaihd.net/i/open/20xxxxxx/2049xxxx/204952xx/20495279/5a30e587ea29838b8cc56e8f7c82d233_20495279_269270_20180222T213434_,150,300,600,1200,2400,.mp4.csmil/master.m3u8?__b__=2000&hdnts=st=1519667985~exp=3667151632~acl=/*~hmac=2e33027cf399db818ca38ef80d645ef3dbde1ed1ac3f773f04000b3c26a48386&cc1=name=Svenska%20f%C3%B6r%20h%C3%B6rselskadade~default=no~forced=no~lang=sv~uri=https%3A%2F%2Fsubstitch.play.mtgx.tv%2Fsubtitle%2Fconvert%2Fvtt%3Fsource%3Dhttps%3A%2F%2Fcdn-subtitles-mtgx-tv.akamaized.net%2Fpitcher%2F20xxxxxx%2F2049xxxx%2F204952xx%2F20495279%2F2800120619%2Funified.subs.20495279_sv_sdh.vtt%26output%3Dm3u8"
```

Reply:
Before i complained that `options.segments` was not defined